src/exoticatechnologies/campaign/listeners/CampaignEventListener.kt

```diff
@@ -48,6 +48,7 @@ class CampaignEventListener(permaRegister: Boolean) : BaseCampaignEventListener(
     override fun reportShownInteractionDialog(dialog: InteractionDialogAPI) {
         val interactionTarget = dialog.interactionTarget ?: return
         val dialogPlugin = dialog.plugin
+        val memoryMap = dialog.plugin?.memoryMap ?: return //0.98a fix
```

Comment:
wouldn't something like this make more sense to avoid early-returns?
```
val memoryMap = dialog.plugin?.memoryMap ?: mapOf()
```

also get rid of the inline comment.